src/lib/ares_init.c

```diff
@@ -1472,7 +1472,57 @@ static int init_by_resolv_conf(ares_channel channel)
     /* Catch the case when all the above checks fail (which happens when there
        is no network card or the cable is unplugged) */
     status = ARES_EFILE;
+#elif defined(__MVS__)
+
+  static struct __res_state *res = 0;
+  int count4, count6;
+  __STATEEXTIPV6 *v6 = res->__res_extIPv6;
```

Comment:
pretty sure you need to set   v6=res->__res_extIPv6;  after setting res as otherwise you're dereferencing a NULL ptr here.